### PR TITLE
[WIP]: Remove lxml and replace with buildin xml

### PR DESCRIPTION
**Pull request by swamper123**
Things this PR shall tackle:

- remove lxml and dependencies (#299 )
- ~~fix missing Nodes after import/create(#307, #308 )~~
- ~~fix missing references after import/create~~
- replace lxml with python buildin xml

~~Optional:
check if Parentnodes already contains a Childnode with same Browsename (see also #306 , maybe the problem can be cought here while importing files as well)~~

~~If you got other problems with xml, except the mentioned ones, please post them here as well, so I can take care of them as well in this PR.~~

### CODE REVIEW

**Comment by oroulet** [comment]
@swamper123 great you are adressing this. But not ry to do to much ion one PR. First remove lxml, make all tests pass.
MAybe add some tests which are importing some well know models (like theDi, and robotics models) then try to fix bugs around

**Comment by swamper123** [comment]
@oroulet yeah you are right about that. I should keep it small first...I hope that whole stuff is not intermeshed much. >_<

File: asyncua/common/structures104.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def clean_name(name):
     newname = re.sub(r'^[0-9]+', r'_\g<0>', newname)
 
     if name != newname:
-        logger.warning("renamed %s to %s due to Python syntax", name, newname)
+        logger.warning(f"renamed {name} to {newname} due to Python syntax")
```
Comment (oroulet) [inline asyncua/common/structures104.py:22]: Do not do that. The f-string stuff should not be used for logging. Try to use pylint for example to avoid such errors

Comment (swamper123) [reply asyncua/common/structures104.py:22]: I can revert that. Why are f-strings in logging bad practice?

Comment (oroulet) [reply asyncua/common/structures104.py]: logging uses lazy string computattion. when using f-string syntax  you force creating the string at runtime, even when not used

**Comment by swamper123** [comment]
So testing with Robotics or DI isn't as easy than be thought. While DI is running (more or less, with a dirty workaround because of `ParameterIdentifier` Node whichs parent is `NameSpaceUri` whoms Value is allowed to be empty. Perhaps this is causing these import errors from the standard_address_spaces?) Robotics is missing some Node attributes, which seems to be missing.
That stuff is even more complicated if there are remaining Errors in the Nodeset which may be fixed in v.1.05. That can mislead the whole parsing thing as well.

We may need to take a look on how the Nodes are thrown together. Perhaps there have been stuff changed? Maybe we refactor some class Names to be spec conform, so it is easier to understand why class xyz exists and where to look at in the specs.

I saw as well that[ *.bsd files are obsolete, s](https://github.com/OPCFoundation/UA-Nodeset/blob/v1.04/readme.md)o the tests should use an _test_file.xml_ instead.

I may have to take a look especially on `make_model_from_string` and `make_model_from_file` since there have been discrepancy (maybe caused by the bsd file as example file).

TLDR: XML Topic is aaaaaaahhhhhhhhh....but I am on it.

File: asyncua/client/client.py [hunk 2]
```diff
@@ -558,7 +558,7 @@ async def create_subscription(self, period, handler):
             params.RequestedLifetimeCount = 10000
             params.RequestedMaxKeepAliveCount = 3000
             params.MaxNotificationsPerPublish = 10000
-            params.PublishingEnabled = True
+            params.PublishingEnabled = publishing
```
Comment (oroulet) [inline asyncua/client/client.py:561]: why do we have that kind of changes in a branch supposed to focus on removing lxml dependency??

File: asyncua/common/subscription.py [hunk 3]
```diff
@@ -370,3 +387,40 @@ def deadband_monitor(self, var, deadband_val, deadbandtype=1, queuesize=0, attr=
...
+        :return: Return a Set Publishing Mode Result
+        """
+        self.logger.info("set_publishing_mode")
+        params = ua.SetPublishingModeParameters()
+        params.SubscriptionIds = [self.subscription_id]
+        params.PublishingEnabled = publishing
+        result = await self.server.set_publishing_mode(params)
+        if result[0].is_good():
+            self.parameters.PublishingEnabled = publishing
+        return result
```
Comment (oroulet) [inline asyncua/common/subscription.py:426]: here too??? try to rebase on master

Comment (swamper123) [reply asyncua/common/subscription.py:426]: Since this is a draft (and it will take some time), you may take a look on it, when it's ready. Otherwise you look through half-baked code.

It will take a while to understand the whole stuff (objectify, xml.etree(if I even use that, don't know if DOM/minidom or SAX be better?), xml-files/Spec thing etc.). It seems that this isn't an easy three liner code to replace, so that tests run fine. 😕

Comment (oroulet) [reply asyncua/common/subscription.py:426]: I just wanted to see if I could help before you spend too much time on it...

Comment (swamper123) [reply asyncua/common/subscription.py]: Sorry if it sounded rude, it wasn't meant like that. I appreciate your help realy much.
I believe that this Draft will stay the way it is for a while, until the underlaying stuff works the way as expected. The rat tail may be to long for one PR. :/

Comment (oroulet) [reply asyncua/common/subscription.py]: no pb
